internal/controller/operator/factory/build/defaults.go

```diff
@@ -582,5 +581,15 @@ func addDefaultsToVMBackup(cr *vmv1beta1.VMBackup, useDefaultResources bool, app
 	}
 
 	cr.Resources = Resources(cr.Resources, config.Resource(appDefaults.Resource), useDefaultResources)
+}
 
+func addVMServiceScrapeDefaults(objI interface{}) {
+	cr := objI.(*vmv1beta1.VMServiceScrape)
+	if cr == nil {
+		return
+	}
+	c := getCfg()
+	if cr.Spec.DiscoveryRole == "" && c.VMServiceScrapeDefault.EnforceEndpointslices {
```

Comment:
Please update service monitor converter as well.

Reply:
@f41gh7 @Haleygo This doesn't solve service monitor conversion problem for which I raised my PR. Please update that as well.

Reply:
@ajaychoudhary-hotstar We don't need to update converter specificly, the workflow is:
```
user create servicemonitor-->vm-operator convert to vmservicescrape--> 
                                                                       vm-operator generate scrape config from vmservicescrape for vmagent
user create vmservicescrape --------------------------------------------> 
```
So we just need to change the default discovery role to `endpointslices` for all the vmservicescrape when generate the scrape config for vmagent, it doesn't matter if the vmservicescrape is created directly or converted.
You can test it using this branch.

Reply:
Thanks for the confirmation. What I noticed in our cluster is it doesn't have discoveryRole: endpoints by default present in any of the vmservicescrape objects. Based on the current implementation it should be added for the converted servicescape objects also right?

Reply:
>What I noticed in our cluster is it doesn't have discoveryRole: endpoints by default present in any of the vmservicescrape objects.

It's not added to vmservicescrape objects directly, otherwise, vm-operator has to make redundant update for vmservicescrape. It's added in vmagent scrape_config for job generated from vmservicescrape.

>Based on the current implementation it should be added for the converted servicescape objects also right?

yes, it will apply to all the vmservicescrape that without `discoveryRole` specified.

Reply:
>But it'd be great to have some test for it.

@f41gh7 yeah, I wanted that but failed, it's more like an integration test.